vmdb/app/models/provider_foreman.rb

```diff
@@ -0,0 +1,18 @@
+class ProviderForeman < Provider
+  has_one :configuration_manager, :class_name => "ConfigurationManagerForeman", :dependent => :destroy
+  has_one :provisioning_manager,  :class_name => "ProvisioningManagerForeman",     :dependent => :destroy
+
+  def before_save
+    build_provisioning_manager unless provisioning_manager
+    build_configuration_manager unless configuration_manager
+  end
+
+  def connection_attrs
```

Comment:
Can you take an auth_type parameter here which defaults to nil or maybe an options hash, and then pass auth_type to authentication_userid and authentication_password?  If we will reuse this method in the future, then it's required by most of the other .connect methods, for example with [Amazon](https://github.com/ManageIQ/manageiq/blob/master/vmdb/app/models/ems_amazon.rb#L37-L44).  In addition, I wonder if credential validity should be checked here as well as in that link.